### PR TITLE
Allow easy status-go log level modification via feature flag (#2756)

### LINKED ISSUE #2756: Allow easy status-go log level modification
### User Story

As a developer, I want to easily modify `status-go` log level via feature flags

### Description

*Type*: Feature

`status-go` log level is hardcoded. It should be possible to set it via [feature flag](https://github.com/status-im/status-react/blob/develop/.env)

Log levels are set here:

* android https://github.com/status-im/status-react/blob/develop/modules/react-native-status/android/src/main/java/im/status/ethereum/module/StatusModule.java#L182
*ios https://github.com/status-im/status-react/blob/develop/modules/react-native-status/ios/RCTStatus/RCTStatus.m#L205

### PR DESCRIPTION
**Pull request by pedropombeiro**
Fixes #2756 

### Summary:

The log level passed by the Android and iOS code to status-go was hard-coded to `"INFO"`. This PR uses the log level defined by a feature flag in the `.env` files allowing status-go to get the correct value to initialize a node, taking the correct case conversion into consideration. No change was required on the Go side because the log level is already being read from the node config JSON: https://github.com/status-im/status-go/blob/develop/geth/node/manager.go#L498.
In the process, got rid of superfluous `DEBUG_LOGS_ENABLED` feature flag using the value of `LOG_LEVEL` to keep functionality identical.
...

### Steps to test:
- Open debug build of Status
- Verify that the logs contain `debug` level entries

status: ready

### CODE REVIEW

**Comment by jeluard** [comment]
Wondering if it's worth having a dedicated flag for `status-go` log level. Maybe @rasom or @oskarth have an opinion on that?

**Comment by oskarth** [comment]
@jeluard Possibly. By default the log levels should probably be in sync, but it might be nice to instrument status-go to be less noisy for perf for example, while having status-react be useful for debugging. Don't think it is a requirement though, more nice to have.

**Comment by pedropombeiro** [comment]
Makes sense, I'll add that to the PR then.

**Review by oskarth — APPROVED** [review]
LGTM

**Comment by pedropombeiro** [comment]
Added `LOG_LEVEL_STATUS_GO` to differentiate from the regular `LOG_LEVEL` feature flag used for the React side.
@oskarth: One thing I'd appreciate confirmation of is whether the feature flag used in `ios/StatusIm/AppDelegate.m` should in fact be `LOG_LEVEL`. It seems so, since we're setting the React log threshold, but doesn't hurt to double-check.

**Comment by oskarth** [comment]
@PombeirP I don't actually remember, but I guess the answer is yes. Here's the original PR https://github.com/status-im/status-react/pull/1803 but I forgot what specific problem for QA we were solving. Let's say yes :)

**Comment by pedropombeiro** [comment]
@oskarth Do we want to enable logging by default for the Android bundle, now that the log level can be controlled independently? https://github.com/status-im/status-react/blob/develop/modules/react-native-status/android/src/main/java/im/status/ethereum/module/StatusModule.java#L180

**Comment by jeluard** [comment]
@PombeirP An alternative could be to enable only if a log level has been explicitly set. Also I think `INFO` as default for dev builds is enough as we only care about those logs for very specific scenario.

**Comment by pedropombeiro** [comment]
@jeluard makes total sense. I'll make both changes.

By the way, is logging known to be working on non-rooted Android devices? The current code is logging to a file under `/data/users/0`, but no log file is visible on my Nexus 6P. I had to compile with a different path rooted in `/storage/emulated/0` to start seeing the geth log file.

**Comment by jeluard** [comment]
@PombeirP Not quite sure. @rasom or @flexsurfer might provide some details about that. It's quite possible it is currently broken.

**Comment by jeluard** [comment]
@PombeirP Build fails because of the following error:

```
status/android/src/main/java/im/status/ethereum/module/StatusModule.java:182: error: cannot find symbol
            jsonConfig.put("LogEnabled", !Strings.IsNullOrEmpty(this.logLevel));
                                          ^
  symbol:   variable Strings
  location: class StatusModule
```

**Comment by pedropombeiro** [comment]
Hi @jeluard, sorry about that, not sure how I didn't see it. By the way, it would be nice if Jenkins build logs would be visible to external contributers (unless there's some logistical reason for why they currently aren't).

**Comment by jeluard** [comment]
@PombeirP Indeed this is part of our plans, hopefully we will get it sooner than later.

**Comment by oskarth** [comment]
@jeluard @rasom could you please review this PR?

**Review by jeluard — APPROVED** [review]

File: .env.prod [hunk 1]
```diff
@@ -1,10 +1,10 @@
 TESTFAIRY_ENABLED=0
 NOTIFICATIONS_WIP_ENABLED=1
-DEBUG_LOGS_ENABLED=0
 STUB_STATUS_GO=0
 ETHEREUM_DEV_CLUSTER=0
 MAINNET_NETWORKS_ENABLED=0
 ERC20_ENABLED=0
 OFFLINE_INBOX_ENABLED=0
 LOG_LEVEL=info
+LOG_LEVEL_STATUS_GO=
```
Comment (oskarth) [inline .env.prod:9]: Missed this in review, but noticed now. Seems wrong? @PombeirP

Comment (pedropombeiro) [reply .env.prod:9]: @oskarth That is based on this suggestion: https://github.com/status-im/status-react/pull/2803#issuecomment-354145328

Comment (oskarth) [reply .env.prod:9]: Oh my bad, missed this. Gotcha.

**Comment by oskarth** [comment]
This likely caused https://github.com/status-im/status-react/issues/2925

Log levels should _always_ be set by default, even if it is just FATAL in e.g. FATAL, ERROR, WARN, INFO, DEBUG, TRACE hierarchy. Optionally they can be overwritten, but this way there is graceful and not undefined behavior.

**Comment by oskarth** [comment]
@andytudhope Did QA test this? Why was it merged straight away?

**Comment by andytudhope** [comment]
Can't find the slack thread now @oskarth, but was asked to merge because I had my keys set up correctly and Julien was battling with his still. Apologies!

**Comment by oskarth** [comment]
@andytudhope No worries, just trying to debug process so we can improve until next time :) Let's make sure stuff like this passes through QA in future / is confirmed to have been exercised in configuration space.

**Pull request merged**